app/src/main/java/org/asteroidos/sync/asteroid/AsteroidBleManager.java

```diff
@@ -0,0 +1,177 @@
+                }
+            }
+
+            for (IConnectivityService service : synchronizationService.getServices()) {
+
+                BluetoothGattService bluetoothGattService = gatt.getService(service.getServiceUUID());
+                HashMap<UUID, IConnectivityService.Direction> characteristics = service.getCharacteristicUUIDs();
+
+                if (characteristics == null) {
+                    Log.d(TAG, "CHAR NULL! " + characteristics + " " + service.getServiceUUID());
+                    continue;
+                } else {
+                    Log.d(TAG, "CHAR " + characteristics + " " + service.getServiceUUID());
+                }
+
+                characteristics.forEach((uuid, direction) -> {
+                    BluetoothGattCharacteristic characteristic = bluetoothGattService.getCharacteristic(uuid);
+                    gattChars.put(characteristic, direction);
+                    bluetoothGattService.addCharacteristic(characteristic);
+                });
+
+                gattServices.add(bluetoothGattService);
+            }
+
+            supported = (batteryCharacteristic != null && notify);
+
+            Log.d(TAG, "FOUND? " + supported);
+
+            // Return true if all required services have been found
+            return supported;
+        }
+
+        @Override
+        protected final void initialize() {
+            beginAtomicRequestQueue()
+                    .add(requestMtu(256) // Remember, GATT needs 3 bytes extra. This will allow packet size of 244 bytes.
+                            .with((device, mtu) -> log(Log.INFO, "MTU set to " + mtu))
+                            .fail((device, status) -> log(Log.WARN, "Requested MTU not supported: " + status)))
+                    .done(device -> log(Log.INFO, "Target initialized"))
+                    .fail((device, status) -> Log.e("Init", device.getName() + " not initialized with error: " + status))
+                    .enqueue();
+
+            setNotificationCallback(batteryCharacteristic).with(((device, data) -> setBatteryLevel(data)));
+            readCharacteristic(batteryCharacteristic).with(((device, data) -> setBatteryLevel(data))).enqueue();
+            enableNotifications(batteryCharacteristic).enqueue();
+
+            gattChars.forEach((characteristic, direction) -> {
+                System.out.println(characteristic.toString() + ": " + direction.name());
+                if (direction == IConnectivityService.Direction.RX) {
+                    setNotificationCallback(characteristic).with((device, data)
+                            -> informService(characteristic.getUuid(), data.getValue()));
+
+                    //sometimes crashes with null pointer or sends random data
+                    enableNotifications(characteristic).with((device, data)
+                            -> informService(characteristic.getUuid(), data.getValue())).enqueue();
+                } else if (direction == IConnectivityService.Direction.TX) {
```

Comment:
currently it's either RX or TX so I would use if/else.
But if you just have an empty block I would just remove the else block.